lib/web_console/whiny_object.rb

```diff
@@ -0,0 +1,23 @@
+module WebConsole
+  class WhinyObject < SimpleDelegator
```

Comment:
Since I was not familiar with the `ActiveSupport::Notifications` API, I just decided to play a bit in order to form some opinion.

I think you are thinking on using `ActiveSupport::LogSubscriber` and trigger events with `ActiveSupport::Notifications.instrument`.

I tested all this stuff with changes in the `middleware.rb` file. A working (but still ugly) solution is [here](https://gist.github.com/jmbejar/95854eb3005a78b915c4).

I personally feel this is too much overhead to just render a log message. In fact, looking at the Rails code, this is used just in very few occasions (sql logs, time to render the view). Looks like `ActiveSupport::Notifications` is used when there is the need to really instrument something (I mean, measuring time spent and so on). For this reason, the only way to trigger events is by calling `instrument` method (as far I can say).

Also, we still have to decide where the conditional logic to show the log messages goes, the `ActiveSupport::Notifications` would be just a fancy way to call the logger.

Regarding where the conditional to show logs goes, I see 3 options:

- Have it on the middleware (it was my first approach)
- Have specific classes to log these errors (this was your initial approach and it was I followed)
- Have the logic inside Response and Request (logger can be retrieved from the env in the middleware and passed to these objects in initialization)

I recognize that having 3 `whipy` classes can be a bit of overhead in this case, however I wouldn't go with the first option. The second option seems feasible and I could try it if you want.